components/eventing-controller/controllers/subscription/nats/unit_test.go

```diff
@@ -457,76 +458,45 @@ func setupTestEnvironment(t *testing.T) *TestEnvironment {
 	}
 }
 
-func createFakeClient(g *GomegaWithT) client.WithWatch {
+func createFakeClient(t *testing.T) client.WithWatch {
 	testEnv = &envtest.Environment{
```

Comment:
I removed this assignment and the tests looked good. Probably this is not necessary here?

Reply:
I don't really understand what you removed, that the tests remained green ? Here I am registering our eventing crds including subscription into the testing scheme. Without that you get something like `no kind is registered for the type v1alpha1.Subscription`

Reply:
Yes, you are right, `testEnv` is not needed here, as these are unit tests, we mock here everything and we are not using k8s API server